# **ACTIVIDAD 1**

In [8]:
pip install numpy

## 1. Creación Dataframe

In [1]:
import pandas as pd
df_url = pd.read_csv('https://raw.githubusercontent.com/selva86/datasets/master/BostonHousing.csv')

A través de la función `pd.read_csv` importamos el Dataframe de la url indicada. 

Para hacernos una idea de la estructura del Dataframe, utilizamos el print(df)

In [2]:
print(df_url)

        crim    zn  indus  chas    nox     rm   age     dis  rad  tax  \
0    0.00632  18.0   2.31     0  0.538  6.575  65.2  4.0900    1  296   
1    0.02731   0.0   7.07     0  0.469  6.421  78.9  4.9671    2  242   
2    0.02729   0.0   7.07     0  0.469  7.185  61.1  4.9671    2  242   
3    0.03237   0.0   2.18     0  0.458  6.998  45.8  6.0622    3  222   
4    0.06905   0.0   2.18     0  0.458  7.147  54.2  6.0622    3  222   
..       ...   ...    ...   ...    ...    ...   ...     ...  ...  ...   
501  0.06263   0.0  11.93     0  0.573  6.593  69.1  2.4786    1  273   
502  0.04527   0.0  11.93     0  0.573  6.120  76.7  2.2875    1  273   
503  0.06076   0.0  11.93     0  0.573  6.976  91.0  2.1675    1  273   
504  0.10959   0.0  11.93     0  0.573  6.794  89.3  2.3889    1  273   
505  0.04741   0.0  11.93     0  0.573  6.030  80.8  2.5050    1  273   

     ptratio       b  lstat  medv  
0       15.3  396.90   4.98  24.0  
1       17.8  396.90   9.14  21.6  
2       17.8  3

## 2. Obtener las dimensiones de dicho Dataframe 

In [3]:
df_url.shape

(506, 14)

A través de la función `shape` podemos observar de manera simplificada que nuestro Dataframe contiene 506 fila y 14 columnas.

## 3. Obtener el resumen estadístico de todas las columnas

In [5]:
df_url.describe(include='all')

crim          zn       indus        chas         nox          rm  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean     3.613524   11.363636   11.136779    0.069170    0.554695    6.284634   
std      8.601545   23.322453    6.860353    0.253994    0.115878    0.702617   
min      0.006320    0.000000    0.460000    0.000000    0.385000    3.561000   
25%      0.082045    0.000000    5.190000    0.000000    0.449000    5.885500   
50%      0.256510    0.000000    9.690000    0.000000    0.538000    6.208500   
75%      3.677083   12.500000   18.100000    0.000000    0.624000    6.623500   
max     88.976200  100.000000   27.740000    1.000000    0.871000    8.780000   

              age         dis         rad         tax     ptratio           b  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean    68.574901    3.795043    9.549407  408.237154   18.455534  356.674032   
std     28.148861    2.105710    8.707259  168.537116    2.164946   91.294864   
min      2.900000    1.129600    1.000000  187.000000   12.600000    0.320000   
25%     45.025000    2.100175    4.000000  279.000000   17.400000  375.377500   
50%     77.500000    3.207450    5.000000  330.000000   19.050000  391.440000   
75%     94.075000    5.188425   24.000000  666.000000   20.200000  396.225000   
max    100.000000   12.126500   24.000000  711.000000   22.000000  396.900000   

            lstat        medv  
count  506.000000  506.000000  
mean    12.653063   22.532806  
std      7.141062    9.197104  
min      1.730000    5.000000  
25%      6.950000   17.025000  
50%     11.360000   21.200000  
75%     16.955000   25.000000  
max     37.970000   50.000000

A través de la función `.describe(inclued='all')`, podemos ver un resumen de los estadísticos básicos para nuestro Dataframe en todas las columnas que conforman el Dataframe.

## 4. Crear un segundo Dataframe a partir del fichero .csv solo con las columnas ‘crim’ y ‘medv’.  (Importante: No debe venir del Dataframe creado en el apartado 1)

In [9]:
import pandas as pd
df_url1 = pd.read_csv('https://raw.githubusercontent.com/selva86/datasets/master/BostonHousing.csv', usecols=['crim', 'medv'])

Para que la creación no sea a partir de nuestro Dataframe creado en el primer apartado, importamos nuevamente el fichero csv y con la función `df_selected`, indicamos las columnas que queremos para nuestro nuevo Dataframe. 

Utilizamos print para comprobar de manera visual el cósigo ejecutado anteriormente.

In [10]:
print(df_url1)

        crim  medv
0    0.00632  24.0
1    0.02731  21.6
2    0.02729  34.7
3    0.03237  33.4
4    0.06905  36.2
..       ...   ...
501  0.06263  22.4
502  0.04527  20.6
503  0.06076  23.9
504  0.10959  22.0
505  0.04741  11.9

[506 rows x 2 columns]


# **ACTIVIDAD 2**

*Creamos un DataFrame utilizando el fichero de la Url.*

In [19]:
import pandas as pd
df_url = pd.read_csv('https://raw.githubusercontent.com/selva86/datasets/master/Cars93_miss.csv')                     

* Hacemos una pequeña revisión de la estructura del Dataframe

In [20]:
print(df_url)

   Manufacturer    Model     Type  Min.Price  Price  Max.Price  MPG.city  \
0         Acura  Integra    Small       12.9   15.9       18.8      25.0   
1           NaN   Legend  Midsize       29.2   33.9       38.7      18.0   
2          Audi       90  Compact       25.9   29.1       32.3      20.0   
3          Audi      100  Midsize        NaN   37.7       44.6      19.0   
4           BMW     535i  Midsize        NaN   30.0        NaN      22.0   
..          ...      ...      ...        ...    ...        ...       ...   
88   Volkswagen  Eurovan      Van       16.6   19.7       22.7      17.0   
89   Volkswagen   Passat  Compact       17.6   20.0       22.4      21.0   
90   Volkswagen  Corrado   Sporty       22.9   23.3       23.7      18.0   
91        Volvo      240  Compact       21.8   22.7       23.5      21.0   
92          NaN      850  Midsize       24.8   26.7       28.5      20.0   

    MPG.highway             AirBags DriveTrain  ... Passengers  Length  \
0          31

## 1. Revisión valores nulos en el Dataframe. El comando que se utilice para este propósito deberá devolver únicamente un resultado binario, es decir, si hay o no presente nulos en el Dataframe

In [51]:
Valores_nulos = df_url.isnull().values.any()
print(Valores_nulos)


True


Revisión de valores nulos. Si es TRUE si que hay valores nulos, si es FALSE, no hay valores nulos.

## 2. Contar el número de nulos en cada columna

In [26]:
df_url.isnull().sum()

Manufacturer           4
Model                  1
Type                   3
Min.Price              7
Price                  2
Max.Price              5
MPG.city               9
MPG.highway            2
AirBags               38
DriveTrain             7
Cylinders              5
EngineSize             2
Horsepower             7
RPM                    3
Rev.per.mile           6
Man.trans.avail        5
Fuel.tank.capacity     8
Passengers             2
Length                 4
Wheelbase              1
Width                  6
Turn.circle            5
Rear.seat.room         4
Luggage.room          19
Weight                 7
Origin                 5
Make                   3
dtype: int64

Número de valores nulos en cada columna

## 3. ¿Cuál es el valor máximo de los precios (Columna ‘Price’) de un carro?

In [34]:
max_Price = df_url['Price'].max()
print(max_Price)

61.9


Valor máximo de la columna 'Price' es 61.9

## 4. Obtener, en un solo paso, los valores del fabricante, modelo y tipo ('Manufacturer','Model' y 'Type') del carro de mayor precio (obtenido en el apartado 3)

In [52]:
carro_mayor_precio = df_url.loc[df_url['Price'].idxmax(), ['Manufacturer', 'Model', 'Type']]

print(carro_mayor_precio)


Manufacturer    Mercedes-Benz
Model                    300E
Type                  Midsize
Name: 58, dtype: object


Se muestran los valores de las columnas `'Manufacturer'`, `'Model'`,`'Type'` en función del valor máximo de la columna `'Price'`.

Por si la duda, mostramos todo el Dataframe para comprobar que los códigos ejectados anteriormente son correctos.

In [48]:
pd.set_option('display.max_rows', None)  # Mostrar todas las filas
pd.set_option('display.max_columns', None)  # Mostrar todas las columnas

print(df_url)


     Manufacturer           Model     Type  Min.Price  Price  Max.Price  \
0           Acura         Integra    Small       12.9   15.9       18.8   
1             NaN          Legend  Midsize       29.2   33.9       38.7   
2            Audi              90  Compact       25.9   29.1       32.3   
3            Audi             100  Midsize        NaN   37.7       44.6   
4             BMW            535i  Midsize        NaN   30.0        NaN   
5           Buick         Century  Midsize       14.2   15.7       17.3   
6           Buick         LeSabre    Large       19.9   20.8        NaN   
7           Buick      Roadmaster    Large       22.6   23.7       24.9   
8           Buick         Riviera  Midsize       26.3   26.3       26.3   
9        Cadillac         DeVille    Large       33.0   34.7       36.3   
10       Cadillac         Seville  Midsize       37.5   40.1       42.7   
11      Chevrolet        Cavalier  Compact        8.5   13.4       18.3   
12      Chevrolet        